network/p2p/tracer/gossipSubMeshTracer.go

```diff
@@ -139,6 +151,15 @@ func (t *GossipSubMeshTracer) Prune(p peer.ID, topic string) {
 	lg.Info().Hex("flow_id", logging.ID(id.NodeID)).Str("role", id.Role.String()).Msg("pruned peer")
 }
 
+// SendRPC is called when a RPC is sent. Currently, the GossipSubMeshTracer tracks iHave RPC messages that have been sent.
+// This function can be updated to track other control messages in the future as required.
+func (t *GossipSubMeshTracer) SendRPC(rpc *pubsub.RPC, _ peer.ID) {
```

Comment:
I'd like to propose a refinement for consideration in a separate PR, and you're welcome to open an issue and address it independently:

The current setup involves the `SendRPC` function being invoked by GossipSub in a blocking manner. This implies that any processing performed here will result in a temporary pause of outgoing RPCs. This pause can negatively impact the performance and correctness of GossipSub message dissemination, potentially leading to dropped outgoing RPCs due to a 'context deadline exceeded' error that results from blocking the processing of an RPC. In addition, this can create lock contention between tracking sent RPCs and evaluating incoming RPCs when checking if an `iWant` corresponds to a previously advertised `iHave`.

To mitigate these issues, I suggest implementing a non-blocking approach for processing RPCs. This can be accomplished by utilizing a worker queue and dispatching a copy of the RPC to the queue for further processing while promptly returning from the function. This strategy also eliminates the aforementioned lock contention.

You might draw inspiration from the following example when implementing this solution: https://github.com/onflow/flow-go/blob/master/network/alsp/manager/manager.go#L228-L258.

By reducing contention, we can minimize the risk of deadlocks and improve the overall response time of the system.

Reply:
Follow up issue https://github.com/dapperlabs/flow-internal/issues/1899